workitem/workitem_repository.go

```diff
@@ -193,12 +192,12 @@ func (r *GormWorkItemRepository) FindSecondItem(order *float64, secondItemDirect
 	Item := WorkItem{}
 	var tx *gorm.DB
 	switch secondItemDirection {
-	case DirectionBelow:
-		// Finds the item below which reorder item has to be placed
-		tx = r.db.Where("execution_order < ?", order).Order("execution_order desc", true).Last(&Item)
-
 	case DirectionAbove:
 		// Finds the item above which reorder item has to be placed
+		tx = r.db.Where("execution_order < ?", order).Order("execution_order desc", true).Last(&Item)
```

Comment:
Please improve this query to not sort all WIs with an execution order smaller than `?`. Something like this should do it:

```sql
SELECT *
FROM work_items
WHERE id = (SELECT max(id) FROM work_items WHERE execution_order < ?))
```

But make sure you use `WorkItem{}.TableName()` instead of just `work_items`.